addons/flags/functions/fnc_placeFlag.sqf

```diff
@@ -0,0 +1,79 @@
+#include "..\script_component.hpp"
+/*
+ * Author: Timi007
+ * Starts the placing process of the flag for the player.
+ * Flags can be placed with the special flag items.
+ *
+ * Arguments:
+ * 0: Player <OBJECT>
+ * 1: Flag item <STRING>
+ *
+ * Return Value:
+ * Nothing
+ *
+ * Example:
+ * [player, "ace_flags_white"] call ace_flags_fnc_placeFlag
+ *
+ * Public: No
+ */
+
+params ["_player", "_item"];
+TRACE_2("Placing flag",_player,_item);
+
+(GVAR(flagItemCache) get _item) params ["_flagName", "_texture", "_carrierClass"];
+
+// Set flag start height
+GVAR(objectHeight) = MIN_HEIGHT;
+
+GVAR(isPlacing) = PLACE_WAITING;
+
+private _flag = _carrierClass createVehicle [0, 0, 0];
+_flag setFlagTexture _texture;
+
+// Add info dialog for the player which show the controls
+private _placeFlagText = format [LLSTRING(Place), _flagName];
+[_placeFlagText, LELSTRING(Common,Cancel), LLSTRING(AdjustHeight)] call EFUNC(interaction,showMouseHint);
+
+private _mouseClickID = [_player, "DefaultAction", {
+    GVAR(isPlacing) isEqualTo PLACE_WAITING
+}, {
+    GVAR(isPlacing) = PLACE_APPROVE
+}] call EFUNC(common,addActionEventHandler);
+
+[{ // Start of PFH
+    params ["_args", "_handle"];
+    _args params ["_player", "_item", "_flag", "_mouseClickID"];
+
+    if (isNull _flag || {!([_player, _flag] call EFUNC(common,canInteractWith))}) then {
+        GVAR(isPlacing) = PLACE_CANCEL;
+    };
+
+    if (GVAR(isPlacing) isNotEqualTo PLACE_WAITING) exitWith {
+        [_handle] call CBA_fnc_removePerFrameHandler;
+        call EFUNC(interaction,hideMouseHint);
+        [_player, "DefaultAction", _mouseClickID] call EFUNC(common,removeActionEventHandler);
+
+        if (GVAR(isPlacing) isEqualTo PLACE_APPROVE) then {
+            // End position of the flag
+
+            GVAR(isPlacing) = PLACE_CANCEL;
+
+            [_player, "PutDown"] call EFUNC(common,doGesture);
+
+            _player removeItem _item;
+
+            private _jipID = [QGVAR(flagPlaced), [_player, _item, _flag]] call CBA_fnc_globalEventJIP;
+            [_jipID, _flag] call CBA_fnc_removeGlobalEventJIP;
```

Comment:
Not a huge fan of doing this to add interactions.
However, I'm not sure if using a class XEH would be better. Thoughts @LinkIsGrim ?

Reply:
I think we just need a map of (vehicles->flag items)
then we should be able to add a single pickup class-action on the base

I'm happy to do this if that sounds ok

Reply:
@PabstMirror Can you please check whether the new implementation is as you envisioned it?